src/app/home-page/widget/starred-box/starred-box.component.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { AlertService } from 'app/shared/alert/state/alert.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+import { Event, UsersService, EventsService } from 'app/shared/openapi';
+import { finalize, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-starred-box',
+  templateUrl: './starred-box.component.html',
+  styleUrls: ['./starred-box.component.scss'],
+})
+export class StarredBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  totalStarredWorkflows: number = 0;
+  totalStarredTools: number = 0;
+  totalStarredServices: number = 0;
+  totalStarredOrganizations: number = 0;
+  events: Array<Event> = [];
+  public isLoading = true;
+  EventType = Event.TypeEnum;
+  readonly supportedEventTypes = [
+    Event.TypeEnum.ADDVERSIONTOENTRY,
+    Event.TypeEnum.CREATECOLLECTION,
+    Event.TypeEnum.ADDTOCOLLECTION,
+    Event.TypeEnum.PUBLISHENTRY,
+    Event.TypeEnum.UNPUBLISHENTRY,
+    Event.TypeEnum.MODIFYCOLLECTION,
+    Event.TypeEnum.ADDUSERTOORG,
+  ];
+
+  constructor(private usersService: UsersService, private eventsService: EventsService, private alertService: AlertService) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.getMyEvents();
+    this.usersService.getStarredServices().subscribe((starredServices) => {
+      this.totalStarredServices = starredServices.length;
+    });
+    this.usersService.getStarredTools().subscribe((starredTools) => {
+      this.totalStarredTools = starredTools.length;
+    });
+    this.usersService.getStarredWorkflows().subscribe((starredWorkflows) => {
+      this.totalStarredWorkflows = starredWorkflows.length;
+    });
+    this.usersService.getStarredOrganizations().subscribe((starredOrganizations) => {
+      this.totalStarredOrganizations = starredOrganizations.length;
+    });
+  }
+
+  getMyEvents() {
```

Comment:
Looks like it can be private.

FYI, on my requests to make this an the other field private: It may seem trivial and not worth it, and it won't change the way your code runs. I guess it is trivial, but adding `private` makes it easier to ready the code.

If I see `private getMyEvents()`, I know the method is only called from with this class. But because it is public, I had to double-check to see if you were invoking it from the HTML -- because if you were calling it from somewhere else, you would need to reinitialize `this.events` (`this.events` would keep growing).